src/main/java/at/hannibal2/skyhanni/features/garden/tracking/FarmingTracker.kt

```diff
@@ -0,0 +1,303 @@
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.InformationType
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Medal
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Medal.BRONZE_MEDAL
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Medal.DIAMOND_MEDAL
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Medal.GOLD_MEDAL
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Medal.PLATINUM_MEDAL
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Medal.SILVER_MEDAL
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.MessageType.EDITED_MESSAGE
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.MessageType.NEW_MESSAGE
+import at.hannibal2.skyhanni.config.features.garden.TrackingConfig.Pet
+import at.hannibal2.skyhanni.data.Embed
+import at.hannibal2.skyhanni.data.Field
+import at.hannibal2.skyhanni.data.Footer
+import at.hannibal2.skyhanni.data.Mayor
+import at.hannibal2.skyhanni.data.MayorAPI
+import at.hannibal2.skyhanni.data.PetAPI
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.data.Thumbnail
+import at.hannibal2.skyhanni.data.model.TabWidget
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.events.TabListUpdateEvent
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.features.garden.contest.FarmingContestAPI
+import at.hannibal2.skyhanni.features.garden.farming.GardenCropSpeed
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.round
+import at.hannibal2.skyhanni.utils.RegexUtils.matchAll
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.TabListData
+import at.hannibal2.skyhanni.utils.WebhookUtils
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.minutes
+
+@SkyHanniModule
+object FarmingTracker {
+
+    val storage get() = ProfileStorageData.playerSpecific
+    val config get() = SkyHanniMod.feature.garden.tracking
+
+    var lastNotification = SimpleTimeMark.farPast()
+    var farmingSince = SimpleTimeMark.farFuture()
+    var playerFaceURL = ""
+    var stats = mutableMapOf<String, Int>()
+    var lastTablist = listOf<String>()
+    var cookieBuffTimer = ""
+    var godPotionTimer = ""
+    var activeAnitaBuff = ""
+    var currentCrop: Crop? = null
+    var currentPlacement = 0.0
+    const val SKYHANNI_URL = "https://cdn.discordapp.com/attachments/1094190239532208228/1175194612550344784/bbfwfK0.png?" +
+        "ex=669cdd05&is=669b8b85&hm=f338a9e8536ab9679f91dd1322d2d3d11549c33fb702c2b08d14132a20d860ad&"
```

Comment:
Will this expire or is it fine because of it being in the webhook?

Reply:
I got it from #contrb-chat where it has been since 17th november 2023